mopidy/files/library.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import unicode_literals
+
+import logging
+import operator
+import os
+import stat
+import sys
+import urllib2
+
+
+from mopidy import backend, exceptions, models
+from mopidy.audio import scan, utils
+from mopidy.internal import path
+
+logger = logging.getLogger(__name__)
+
+
+class FilesLibraryProvider(backend.LibraryProvider):
+    """Library for browsing local files."""
+    # TODO: get_images that can pull from metadata and/or .folder.png etc?
+    # TODO: handle playlists?
+
+    @property
+    def root_directory(self):
+        if not self._media_dirs:
+            return None
+        elif len(self._media_dirs) == 1:
+            localpath = self._media_dirs[0]['path']
+            uri = path.path_to_uri(localpath)
+        else:
+            uri = u'file:root'
+        return models.Ref.directory(name='Files', uri=uri)
+
+    def __init__(self, backend, config):
+        super(FilesLibraryProvider, self).__init__(backend)
+        self._media_dirs = []
+        for entry in config['files']['media_dir']:
+            media_dir = {}
+            media_dict = entry.split(':')
+            try:
+                local_path = path.expand_path(
+                    media_dict[0].encode(sys.getfilesystemencoding()))
+            except:
+                pass
+            if not local_path:
+                logger.warn('Could not expand path %s' % media_dict[0])
+                continue
+            else:
+                try:
+                    st = os.stat(local_path)
+                except:
+                    logger.warn('Could not open %s' % local_path)
+                    continue
+                if not stat.S_ISDIR(st.st_mode):
```

Comment:
Since the only thing `st` is used for is this, keep to the simpler `os.path.isdir(path)`.